### DISCUSSION TITLE
Plugin approach

### DISCUSSION THREAD

**Discussion by jayfk** [cookiecutter/cookiecutter-django#3483] [Ideas]
As Cookiecutter Django gets more and more features, it might be a good idea to pull some of these out into plugins. 

This prevents bloat in the main project and makes life easier for the maintainers.

In order to make things easier to discover we could add a `plugins.json` file containing a curated list of plugins to use.

The prompt would then look something like this:

```
[snip]
use_docker [y]: n
install_plugins [n]: y
# this is where plugin.json kicks in and presents all available plugins
use_heroku [n]: y
use_compressor [n]: y
use_lets_encrypt [n]: n
use_pycharm [n]: y
```

Discuss!

**Comment by ssteinerx** [comment]
I've started working on this in a separate project though I've only started writing documentation so far.  I usually write the docs first to give me a chance to think about the problem before I start writing code.

https://github.com/ssteinerx/cookiecutter-django-influx

It's named 'influx' 'cause the eventual name is "in flux."

**Comment by jangeador** [comment]
@ssteinerx I like the name `plugin` better than `mixin`. django already has`mixins` and it could be confusing. I was reading the notes on the django channels integration pr where you started discussing this feature and it took me a while to realize that you were not really talking about django mixins.

**Comment by ssteinerx** [comment]
The project is now named https://github.com/ssteinerx/cookiecutter-django-flux.

It is the goo that makes welding https://github.com/pydanny/cookiecutter-django/ projects to external dependencies less painful.

**Comment by ssteinerx** [comment]
@jayfk I've now decided on the name.  It makes perfect sense -- flux is a compound that makes welding otherwise incompatible things possible with a firm weld as the result.

**Comment by jangeador** [comment]
So what is next @ssteinerx ? Do you need any help ?

**Comment by ssteinerx** [comment]
I wanted to let this sit a bit as I work on an actual project based on the cookiecutter-django layout in which I need to use React.

**Comment by jayfk** [comment]
@ssteinerx I'd like to work with you on this. I'm about to deploy a Django project with channels support. I think that'd be a good fit for a `flux`/`mixin`/`plugin`.

What do you think?

**Comment by ssteinerx** [comment]
@jayfk Love to!  I've also got a project in the same state.  This would be a great time to get the basic structure in for a flux (reconsidering that due to React Flux)/ mixin/ plugin.  Maybe it should just be called Django-chips (like chocolate chips, but for cookiecutter-django)?

**Comment by ssteinerx** [comment]
@jayfk I'm about done messing with my prototype and am about to start fresh with a new cookiecutter installation and need to add channels into a brand new project.  Did you ever get started on this?
